### PR TITLE
VB-2708 Tidy integration tests

### PR DESCRIPTION
**Pull request by tpmcgowan**
* Make integration tests and helper classes/stubs more consistent in terms of naming, etc
* Where multiple tests check different elements on a page, combine these to a single test
* Remove unnecessary tests (because they are adequately covered by unit tests)
* Improve some selector choices
* New tests for copying and updating session templates

### CODE REVIEW

File: server/routes/prisons/sessionTemplates/singleSessionTemplateController.ts [hunk 1]
```diff
@@ -20,7 +21,7 @@ export default class SingleSessionTemplateController {
       )
 
       const requestVisitStatsDto: RequestSessionTemplateVisitStatsDto = {
-        visitsFromDate: new Date().toISOString().slice(0, 10),
+        visitsFromDate: format(new Date(), 'yyyy-MM-dd'),
```
Comment (tpmcgowan) [inline server/routes/prisons/sessionTemplates/singleSessionTemplateController.ts:24]: Replaced several instances of this to make it clearer what's going on.

File: integration_tests/e2e/prisons/sessionTemplates/addSessionTemplate.cy.ts [hunk 2]
```diff
@@ -1,214 +1,101 @@
...
+    cy.task('stubGetPrison', TestData.prisonDto())
   })
 
-  it('when on add session template page location groups are available', () => {
-    // Given
-    const addSessionTemplatePage = Page.createPage(AddSessionTemplatePage)
-
-    // When
-    addSessionTemplatePage.goTo(prisonCode)
+  it('should add a session template from the listings page', () => {
```
Comment (tpmcgowan) [inline integration_tests/e2e/prisons/sessionTemplates/addSessionTemplate.cy.ts:20]: Consolidated lots of individual tests for specific form fields to one test that checks them in one go.

File: integration_tests/pages/page.ts [hunk 3]
```diff
@@ -1,24 +1,18 @@
 export type PageElement = Cypress.Chainable<JQuery>
 
 export default abstract class Page {
-  static createPage<T>(constructor: new () => T): T {
+  static verifyOnPage<T>(constructor: new () => T): T {
```
Comment (tpmcgowan) [inline integration_tests/pages/page.ts:4]: Making these helper methods same as in our other UI apps for consistency

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**